subprojects/logging/src/main/java/org/gradle/api/logging/Logger.java

```diff
@@ -24,6 +24,11 @@
  * Logging#getLogger(String)}. A {@code Logger} instance is also available through {@link
  * org.gradle.api.Project#getLogger()}, {@link org.gradle.api.Task#getLogger()} and {@link
  * org.gradle.api.Script#getLogger()}.</p>
+ * <br/>
+ * <p><b>CAUTION!</b>
+ * For publicly published plugins, logging sensitive information (credentials, tokens, certain environment variables), at log levels above {@link Logger#debug} is a security vulnerability.
+ * Build logs for Public CI (eg. Travis CI, Circle CI, GitHub Actions) are world-viewable and can expose this sensitive information.
+ * </p>
```

Comment:
WDYT?
> Logging sensitive information (credentials, tokens, certain environment variables) can create security vulnerabilities. 
Logs for build services (e.g. Travis CI, Circle CI, GitHub Actions) are world-viewable and can expose this sensitive information.

Reply:
I intentionally chose 'is a security vulnerability' because the reality is that it truly is a vulnerability. Maybe that information is better conveyed elsewhere?

I do want to be definitive about this so people can reference back to us as an authoritative source on this.

Reply:
We could link to the user manual and provide a longer explanation there.

e.g.,
> Logging sensitive information (credentials, tokens, certain environment variables) is a security vulnerability.  See [our recommendations](link to docs) when logging ... blah blah.

Links to the user manual have to be absolute (https://docs.gradle.org/current/userguide/...)